src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java

```diff
@@ -0,0 +1,133 @@
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
+import static uk.gov.hmcts.reform.sscs.helper.mapping.HearingsCaseMapping.isInterpreterRequired;
+
+
+
+public final class HearingsDurationMapping {
+    public static final int DURATION_SESSIONS_MULTIPLIER = 165;
+    public static final int DURATION_DEFAULT = 30;
+    public static final int MIN_HEARING_DURATION = 30;
+    public static final int MIN_HEARING_SESSION_DURATION = 1;
+    public static final String DURATION_TYPE_NON_STANDARD_TIME_SLOT = "nonStandardTimeSlot";
+    public static final String DURATION_TYPE_STANDARD_TIME_SLOT = "standardTimeSlot";
+    public static final String DURATION_UNITS_MINUTES = "minutes";
+    public static final String DURATION_UNITS_SESSIONS = "sessions";
+
+    private HearingsDurationMapping() {
+
+    }
+
+    public static int getHearingDuration(SscsCaseData caseData, ReferenceDataServiceHolder referenceDataServiceHolder) {
+        OverrideFields overrideFields = OverridesMapping.getOverrideFields(caseData);
+
+        if (nonNull(overrideFields.getDuration()) && overrideFields.getDuration().intValue() >= MIN_HEARING_DURATION) {
+            return overrideFields.getDuration().intValue();
+        }
+
+        Integer duration = getHearingDurationAdjournment(caseData);
+        if (isNull(duration)) {
+            duration = getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
+        }
+
+        return nonNull(duration) ? duration : DURATION_DEFAULT;
+    }
+
+    public static Integer getHearingDurationAdjournment(SscsCaseData caseData) {
+
+        if (DURATION_TYPE_NON_STANDARD_TIME_SLOT.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationType())
+            && isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())) {
+            int duration =  Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration());
+
+            if (DURATION_UNITS_SESSIONS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())
+                && duration >= MIN_HEARING_SESSION_DURATION) {
+                return duration * DURATION_SESSIONS_MULTIPLIER;
+            } else if (DURATION_UNITS_MINUTES.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())
+                && duration >= MIN_HEARING_DURATION) {
+
+
+                return duration;
+            }
+        }
+
+        return null;
+    }
+
+    public static Integer getHearingDurationBenefitIssueCodes(SscsCaseData caseData, ReferenceDataServiceHolder referenceDataServiceHolder) {
+        uk.gov.hmcts.reform.sscs.reference.data.model.HearingDuration hearingDuration = referenceDataServiceHolder.getHearingDurations().getHearingDuration(
```

Comment:
Is this full import path required?

Reply:
Yes, this a hangover from particular wrong turn that I made earlier thanks for pointing it out.